control/velocity_controller/include/velocity_controller/delay_compensation.h

```diff
@@ -39,12 +39,13 @@ class DelayCompensator
     auto pred_pose = current_pose;
     pred_pose.pose.position.x += dx;
     pred_pose.pose.position.y += dy;
-    pred_pose.header.stamp += ros::Duration(delay_time);
+    pred_pose.header.stamp =
+      rclcpp::Time(pred_pose.header.stamp) + rclcpp::Duration::from_seconds(delay_time);
```

Comment:
It would be safeter to take a duration as argument rather than casting a double... but it's beyond porting to ros2, so up to you as I don't know where else that would require changes

Reply:
I had the same concern, but it should be done in the refactoring phase after porting.

Reply:
Sorry, I misunderstood your comment.

First, I tried to write like
```
pred_pose.header.stamp += rclcpp::Duration::from_seconds(delay_time);
```
but it couldn't. 

I just changed to the easy way, but do you know the better way to add the `double` duration to the `builtin_interface::Time`?

Reply:
> I just changed to the easy way, but do you know the better way to add the double duration to the builtin_interface::Time?

I looked it up, there is no operator `+=` defined only `+`, the one you use http://docs.ros2.org/latest/api/rclcpp/namespacerclcpp.html#a31b2ba48a94966a93b36de2e52cae2bb

Reply:
> I had the same concern, but it should be done in the refactoring phase after porting.

I agree. we could add a TODO comment as simplest reminder

Reply:
I added TODO comment.